email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java

```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
         if (runtime != null) {
             Set<Map.Entry<String, ObjectValue>> services = registeredServices.entrySet();
             for (Map.Entry<String, ObjectValue> service : services) {
-                runtime.invokeMethodSync(service.getValue(), ON_MESSAGE, null, ON_MESSAGE_METADATA, email, true);
+                invokeMethodSync(service.getValue(), ON_MESSAGE, ON_MESSAGE_METADATA, email, true);
             }
         } else {
             log.error("Runtime should not be null.");
         }
         return true;
     }
 
+    private void invokeMethodSync(ObjectValue object, String methodName, StrandMetadata metadata, Object... args) {
```

Comment:
This method may require to throw an error as an error is thrown inside this method.